src/components/header/modals/ModalPolkasafe.vue

```diff
@@ -0,0 +1,402 @@
+        const index = substrateAccounts.value.findIndex(
+          (it: SubstrateAccount) => it.address === props.currentAccount
+        );
+        return index;
+      } else {
+        return null;
+      }
+    });
+
+    const copyAddress = (address: string): void => {
+      copy(address);
+      store.dispatch('general/showAlertMsg', {
+        msg: t('toast.copyAddressSuccessfully'),
+        alertType: 'copied',
+      });
+    };
+
+    const windowHeight = ref<number>(window.innerHeight);
+    const onHeightChange = (): void => {
+      const adjustment = width.value > screenSize.sm ? 520 : 390;
+      windowHeight.value = window.innerHeight - adjustment;
+    };
+
+    window.addEventListener('resize', onHeightChange);
+    onHeightChange();
+
+    const displayBalance = (balance: string): number => {
+      return truncate(ethers.utils.formatEther(balance || '0'));
+    };
+
+    const handleGetMultisigAccounts = async (): Promise<void> => {
+      multisigAccounts.value = [];
+      isLoadingPolkasafe.value = true;
+      selAccount.value = '';
+      if (!selectedSignatory.value || currentNetworkChain.value !== astarChain.ASTAR) {
+        return;
+      }
+      try {
+        const extensions = await web3Enable('AstarNetwork/astar-apps');
+        const injector = extensions.find((it) => {
+          return selectedSignatory.value && it.name === selectedSignatory.value.source;
+        }) as any;
+
+        if (!injector) {
+          throw Error('injector not found');
+        }
+        const substratePrefix = 42;
+        const signatory = encodeAddress(selectedSignatory.value.address, substratePrefix);
+
+        const setMultisigAccounts = async (c: Polkasafe): Promise<void> => {
+          try {
+            const { data, error } = await c.connectAddress(signatory);
+            if (error) throw Error(error);
+            if (!data.hasOwnProperty('multisigAddresses')) {
+              throw Error('error fetching multisig accounts');
+            }
+
+            //@ts-ignore
+            const filteredMultisigAccounts = data.multisigAddresses.filter((it: MultisigAddress) =>
+              isValidAddressPolkadotAddress(it.address, ASTAR_ADDRESS_PREFIX)
+            );
+            if (filteredMultisigAccounts.length > 0) {
+              const updatedAccountMap = await Promise.all(
+                filteredMultisigAccounts.map(async (it: MultisigAddress) => {
+                  const balance = await fetchNativeBalance({
+                    api: $api as ApiPromise,
+                    address: it.address,
+                  });
+                  return { ...it, balance };
+                }) as MultisigAddress[]
+              );
+              multisigAccounts.value = updatedAccountMap.sort(
+                (a: MultisigAddress, b: MultisigAddress) => Number(b.balance) - Number(a.balance)
+              );
+            }
+          } catch (error) {
+            console.error(error);
```

Comment:
Maybe to just re-throw the error and handle it on calling side. Do the same for the method below